### DISCUSSION TITLE
Adding cosmic ray models to SpacePy

### DISCUSSION THREAD

**Discussion by ChrisSWDavis** [spacepy/spacepy#795] [Design and roadmapping]
Hi, I was wondering if it would be okay if cosmic ray models for the near-Earth region could be added to SpacePy? Cosmic ray spectra are used all the time across the space sector and in several Earth-based contexts too, but as far as I'm aware none of the bigger open-source space Python projects contain any cosmic ray implementations, so I think it be very valuable if they could be added to SpacePy.

A lot of cosmic ray models are actually reasonably straightforward and some already exist in some user-created Python tools, so it shouldn't be too hard to link to some of these in SpacePy. I assume that the main difficulty would be converting the inputs and outputs so that they're in a good standardised format for SpacePy? During my work at Surrey Space Center, me and some colleagues actually converted a mathematical cosmic ray model into a Python package some time ago ([https://github.com/ssc-maire/CosRayModifiedISO](https://github.com/ssc-maire/CosRayModifiedISO)), and I would be happy to add a SpacePy function for accessing the package. There's also an existing Python package called crflux ([https://github.com/mceq-project/crflux](https://github.com/mceq-project/crflux)), which currently contains several high energy cosmic ray models which could also be accessed through SpacePy via a function.

What are your opinions on this, and how this might be implemented?

Thanks!

**Comment by jtniehof** [comment]
Hi Chris! Sorry for the delay in getting to your reply.

We're always happy to integrate further models into empiricals. But it sounds like you're suggesting wrapping the existing packages with spacepy functions. What additional functionality would you suggest be added by the SpacePy wrapper?
